internal/daemon/daemon.go

```diff
@@ -551,6 +551,16 @@ func (d *Daemon) Stop(sigCh chan<- os.Signal) error {
 		return fmt.Errorf("internal error: no Overlord")
 	}
 
+	// Shut down the service manager, stopping all services. Must do this
+	// before overlord.Stop, as it creates a change and waits for the change,
+	// and overlord.Stop calls StateEngine.Stop, which locks, so Ensure would
+	// result in a deadlock.
+	err := d.overlord.ServiceManager().Shutdown()
```

Comment:
To be consistent with the above note about "shut down" versus "shutdown", I guess this should be called `ShutDown`, because function names are usually verbs. I'll see where the refactoring ends up and rename if necessary.

Reply:
Yeah, this was no longer relevant with the refactoring. It's `servstate.StopRunning` now.